zproject/urls.py

```diff
@@ -254,7 +254,7 @@
          'PATCH': 'update_stream_backend',
          'DELETE': 'deactivate_stream_backend'}),
     url(r'^default_streams$', 'rest_dispatch',
-        {'PATCH': 'add_default_stream',
+        {'PUT': 'add_default_stream',
```

Comment:
Can you make this hunk its own commit at the start of your branch?  It's definitely an improvement, but since it's a functional change that could break some clients (I think this is very unlikely that there's a client that depends on this behavior, but it is possible) I think it deserves its own commit.